### ISSUE TITLE
Can you tell me how train_mainfile.csv and test_mainfile.csv are generated? Thank you.

### ISSUE THREAD

**Issue by ShaoDonCui** [Gopi-Durgaprasad/Speech-To-Text#3]
Check the code in [Notebook](https://github.com/Gopi-Durgaprasad/Speech-To-Text/blob/master/link)
This link is no longer valid.

**Comment by ShaoDonCui** [comment]
I did this 2 files by looking at the code. thanks

**Comment by ShaoDonCui** [comment]
The code I have run successfully.
There are a bunch of low-level errors.

**Comment by Gopi-Durgaprasad** [comment]
Yes, can you post those?

**Comment by ShaoDonCui** [comment]
I found a big BUG, the data of each batch is the same.
I output **targets** variable。
The data in targets is the same.


def train_fn(args, train_loader, model, optimizer, criterion):
    model.train()
    losses = AverageMeter()

    t = tqdm(train_loader)
    for i, data in enumerate(t):

        inputs, targets, input_percentages, target_sizes = data
        

        input_sizes = input_percentages.mul_(int(inputs.size(3))).int()
        
        inputs = inputs.to(args.device)

        out, output_sizes = model(inputs, input_sizes)
        out = out.transpose(0, 1) # TxNxH

        float_out = out.float() # ensure float32 for loss
        loss = criterion(float_out, targets, output_sizes, target_sizes).to(args.device)

**Comment by ShaoDonCui** [comment]
For example this one, with bachsize set to 2. The resulting **targets**  is 2 identical encodings.
tensor([ **7, 16, 19, 21,  9, 24,** 10, 21,  9, 28,  2, 13, 13, 28, 19,  2, 15, 28,
        21, 16, 28, 21,  9,  6, 28, 16, 17,  6, 15, 10, 15,  8, 28, 16,  7, 28,
        21,  9,  6, 28, 21,  6, 15, 21, 28, 21, 16, 28, 20,  6,  6, 28, 24,  9,
         2, 21, 28, 14, 10,  8,  9, 21, 28,  3,  6, 28,  2, 14, 10, 20, 20, 28,
         3, 22, 21, 28, 14,  2, 20, 21,  6, 19, 28, 24, 10, 13, 13, 28, 24,  9,
        16, 28, 17,  6,  6, 17,  6,  5, 28, 16, 22, 21, 28,  7, 10, 19, 20, 21,
        28, 15,  6,  6,  5,  6,  5, 28, 15, 16, 28, 14, 16, 19,  6, 28, 21,  9,
         2, 15, 28, 16, 15,  6, 28,  8, 13,  2, 15,  4,  6,  **7, 16, 19, 21,  9,
        24,** 10, 21,  9, 28,  2, 13, 13, 28, 19,  2, 15, 28, 21, 16, 28, 21,  9,
         6, 28, 16, 17,  6, 15, 10, 15,  8, 28, 16,  7, 28, 21,  9,  6, 28, 21,
         6, 15, 21, 28, 21, 16, 28, 20,  6,  6, 28, 24,  9,  2, 21, 28, 14, 10,
         8,  9, 21, 28,  3,  6, 28,  2, 14, 10, 20, 20, 28,  3, 22, 21, 28, 14,
         2, 20, 21,  6, 19, 28, 24, 10, 13, 13, 28, 24,  9, 16, 28, 17,  6,  6,
        17,  6,  5, 28, 16, 22, 21, 28,  7, 10, 19, 20, 21, 28, 15,  6,  6,  5,
         6,  5, 28, 15, 16, 28, 14, 16, 19,  6, 28, 21,  9,  2, 15, 28, 16, 15,
         6, 28,  8, 13,  2, 15,  4,  6], dtype=torch.int32)

**Comment by ShaoDonCui** [comment]
Did you see it? The resulting data is the same.

**Comment by ShaoDonCui** [comment]
I found the problem. **Change sample = batch[0] to sample = batch[x]**

def _collate_fn(batch):
    def func(p):
        return p[0].size(1)

 
    batch = sorted(batch, key=lambda sample: sample[0].size(1), reverse=True)

    longest_sample = max(batch, key=func)[0]
    freq_size = longest_sample.size(0)
    minibatch_size = len(batch)
    max_seqlength = longest_sample.size(1)
    inputs = torch.zeros(minibatch_size, 1, freq_size, max_seqlength)
    input_percentages = torch.FloatTensor(minibatch_size)
    target_sizes = torch.IntTensor(minibatch_size)
    targets = []
    for x in range(minibatch_size):
        sample = batch[x]
        tensor = sample[0]
        target = sample[1]
        seq_length = tensor.size(1)
        inputs[x][0].narrow(1, 0, seq_length).copy_(tensor)
        input_percentages[x] = seq_length / float(max_seqlength)
        target_sizes[x] = len(target)
        targets.extend(target)
    targets = torch.IntTensor(targets)
    
    return inputs, targets, input_percentages, target_sizes
